src/freenet/node/RequestSender.java

```diff
@@ -1681,7 +1699,7 @@ void ackOpennet(final PeerNode next) {
      * Add the peer.
      * @return True only if there was a fatal timeout and the caller should not unlock.
      */
-    private boolean finishOpennet(PeerNode next) {
+    private boolean finishOpennet(final PeerNode next) {
```

Comment:
Yes, you can't access non-final local variables from anonymous inner classes. We need this variable in run() below.